spec/active_hash/relation_spec.rb

```diff
@@ -50,4 +50,24 @@
       expect(array.size).to eq(2)
     end
   end
+
+  describe "colliding methods" do
+    it "should handle attributes named after existing methods" do
+      klass = Class.new(ActiveHash::Base) do
+        self.data = [
+          {
+            id: 1,
+            name: "Aaa",
+            display: true,
+          },
+          {
+            id: 2,
+            name: "Bbb",
+            display: false,
+          },
+        ]
+      end
+      expect(klass.where(display: true).length).to eq(1)
```

Comment:
maybe also add:

```ruby
   def name
     self[:name] + "!"
   end
```

```ruby
expect(klass.where(name: 'Bbb!').length).to eq(1)
```

Reply:
AFAICT this isn't something Active Record supports (see note below).